### PR TITLE
Attach PHDR metadata as AVPacket side data

### PR DESCRIPTION
**Pull request by tomcrugnale-ssimwave**
The previous approach of providing an AVStream specifially for metadata did not work when the video MXF file was an asset belonging to an IMF CPL.  This is due to the fact that the IMF demuxer expects that there is only 1 stream to be consumed from an MXF file.

Instead of changing stream mappings within imfdec and perhaps breaking other things, extract the interleaved PHDR and attach it as side data to the J2K frame it belongs to.  This allows for tone mapping to be more easily applied in the future also.

### CODE REVIEW

File: libavformat/imfdec.c [hunk 1]
```diff
@@ -585,6 +589,13 @@ static int set_context_streams_from_tracks(AVFormatContext *s)
             av_log(s, AV_LOG_ERROR, "Could not copy stream parameters\n");
             return ret;
         }
+
+        ret = av_dict_copy(&asset_stream->metadata, first_resource_stream->metadata, 0);
```
Comment (tcrugnale-imax) [inline libavformat/imfdec.c:593]: Required to copy the global DoVi metadata that belongs to the stream as a whole.

File: libavformat/imfdec.c [hunk 2]
```diff
@@ -984,6 +995,14 @@ static const AVOption imf_options[] = {
         .default_val = {.str = NULL},
         .flags       = AV_OPT_FLAG_DECODING_PARAM,
     },
+    {
```
Comment (tcrugnale-imax) [inline libavformat/imfdec.c:998]: I'm not entirely pleased with this, but it's the easiest way to propagate stream options to a wrapped (inner) demuxer.  This is how it is done in hls.c, for example.

File: libavformat/mxfdec.c [hunk 3]
```diff
@@ -2501,9 +2499,11 @@ static int mxf_add_phdr_metadata_stream(MXFContext* mxf)
         return AVERROR_INVALIDDATA;
     }
     if (!mxf_match_uid(track->sequence->data_definition_ul, mxf_phdr_image_metadata_item,
-                       sizeof(mxf_phdr_image_metadata_item))) {
+                       sizeof(mxf_phdr_image_metadata_item)) &&
+        !mxf_match_uid(track->sequence->data_definition_ul, mxf_phdr_image_metadata_wrapping_frame,
```
Comment (tcrugnale-imax) [inline libavformat/mxfdec.c:2503]: Another change for compatibility with other MXF files I had been testing with.

**Review by colekas-ssim — APPROVED** [review]
Some -1 changes need to be addressed

**Pull request merged**